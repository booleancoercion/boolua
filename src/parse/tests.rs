use insta::assert_debug_snapshot;
use paste::paste;

macro_rules! parse_and_verify {
    ($name:ident, $path:literal) => {
        paste! {
            #[test]
            fn [<parse_and_verify_ $name>]() {
                let source = std::fs::read_to_string($path).unwrap();
                let ast = crate::parse_source(&source).expect("parsing generated errors");
                assert_debug_snapshot!(ast);
            }
        }
    };
}

parse_and_verify!(simple_if, "programs/simple_if.lua");
