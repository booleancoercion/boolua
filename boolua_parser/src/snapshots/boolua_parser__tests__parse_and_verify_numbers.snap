---
source: src/parse/tests.rs
assertion_line: 19
expression: ast

---
Block {
    stmts: [
        Assignment(
            [
                Simple(
                    Name(
                        "a",
                    ),
                ),
            ],
            [
                Binary(
                    Num(
                        Int(
                            5,
                        ),
                    ),
                    Add,
                    Num(
                        Int(
                            10,
                        ),
                    ),
                ),
            ],
        ),
        Assignment(
            [
                Simple(
                    Name(
                        "b",
                    ),
                ),
            ],
            [
                Binary(
                    Num(
                        Int(
                            4,
                        ),
                    ),
                    Exp,
                    Binary(
                        Num(
                            Int(
                                7,
                            ),
                        ),
                        Exp,
                        Num(
                            Int(
                                2,
                            ),
                        ),
                    ),
                ),
            ],
        ),
    ],
    ret: None,
}
